### PR TITLE
New medical history

### PR DESCRIPTION
**Pull request by saket**

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/medicalhistory/MedicalHistoryRepository.kt [hunk 1]
```diff
@@ -6,13 +6,30 @@ import org.simple.clinic.medicalhistory.sync.MedicalHistoryPayload
...
 import org.simple.clinic.sync.SynceableRepository
+import org.threeten.bp.Instant
 import java.util.UUID
 import javax.inject.Inject
 
 class MedicalHistoryRepository @Inject constructor(
     val dao: MedicalHistory.RoomDao
 ) : SynceableRepository<MedicalHistory, MedicalHistoryPayload> {
 
+  // TODO: Android Test.
```
Comment (saket) [inline app/src/main/java/org/simple/clinic/medicalhistory/MedicalHistoryRepository.kt]: Whoops I forgot to do this. I'll add this in my next PR.

File: app/src/main/res/layout/screen_new_medical_history.xml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  android:layout_height="match_parent">
+
+  <android.support.v7.widget.Toolbar
+    android:id="@+id/newmedicalhistory_toolbar"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    app:navigationIcon="@drawable/ic_arrow_back_24dp"
+    app:title="@string/newmedicalhistory_screen_title" />
+
+  <ScrollView
```
[the remark this answers is not in the document]
Comment (saket) [reply app/src/main/res/layout/screen_new_medical_history.xml]: There was a reason with the old design, but not anymore. Considering that we're low on time, I decided not to change things.

File: app/src/main/java/org/simple/clinic/medicalhistory/newentry/NewMedicalHistoryScreen.kt [hunk 3]
```diff
@@ -55,6 +56,13 @@ class NewMedicalHistoryScreen(context: Context, attrs: AttributeSet) : RelativeL
       screenRouter.pop()
     }
 
+    heartAttackQuestionView.render(MedicalHistoryQuestion.HAS_HAD_A_HEART_ATTACK)
```
[the remark this answers is not in the document]
Comment (saket) [reply app/src/main/java/org/simple/clinic/medicalhistory/newentry/NewMedicalHistoryScreen.kt]: Will do

File: app/src/main/java/org/simple/clinic/medicalhistory/newentry/NewMedicalHistoryScreenController.kt [hunk 4]
```diff
@@ -21,17 +24,57 @@ class NewMedicalHistoryScreenController @Inject constructor(
...
+    return events
+        .ofType<MedicalHistoryAnswerToggled>()
+        .filter { it.question != MedicalHistoryQuestion.NONE && it.selected }
+        .map { { ui: Ui -> ui.unSelectNoneAnswer() } }
+  }
+
+  private fun enableSaveButton(events: Observable<UiEvent>): Observable<UiChange> {
+    return events
+        .ofType<MedicalHistoryAnswerToggled>()
+        .scan(0) { selectedAnswersCount, toggleEvent ->
```
[the remark this answers is not in the document]
Comment (saket) [reply app/src/main/java/org/simple/clinic/medicalhistory/newentry/NewMedicalHistoryScreenController.kt]: 😄

File: app/src/main/java/org/simple/clinic/newentry/PatientEntryScreen.kt [hunk 5]
```diff
@@ -159,14 +165,23 @@ class PatientEntryScreen(context: Context, attrs: AttributeSet) : RelativeLayout
...
+  fun preFillFields(entry: OngoingPatientEntry) {
+    fullNameEditText.setTextAndCursor(entry.personalDetails?.fullName)
+    phoneNumberEditText.setTextAndCursor(entry.phoneNumber?.number)
+    dateOfBirthEditText.setTextAndCursor(entry.personalDetails?.dateOfBirth)
+    ageEditText.setTextAndCursor(entry.personalDetails?.age)
+    colonyOrVillageEditText.setTextAndCursor(entry.address?.colonyOrVillage)
+    districtEditText.setTextAndCursor(entry.address?.district)
+    stateEditText.setTextAndCursor(entry.address?.state)
+
+    entry.personalDetails?.gender?.let {
```
[the remark this answers is not in the document]
Comment (saket) [reply app/src/main/java/org/simple/clinic/newentry/PatientEntryScreen.kt]: This screen handles saving and restoring state manually -- I don't recall why :/ 
We'll have to revisit this later.

File: app/src/main/java/org/simple/clinic/medicalhistory/MedicalHistoryQuestionView.kt [hunk 6]
```diff
@@ -11,17 +11,29 @@ import kotterknife.bindView
...
 
+data class MedicalHistoryAnswerToggled(val question: MedicalHistoryQuestion, val selected: Boolean) : UiEvent {
+  override val analyticsName = "New Medical History:Answer Toggled"
+}
+
 @SuppressLint("ClickableViewAccessibility")
 class MedicalHistoryQuestionView(context: Context, attrs: AttributeSet) : RelativeLayout(context, attrs) {
 
   val checkBox by bindView<CheckBox>(R.id.newmedicalhistory_item_checkbox)
+  val toggles = PublishSubject.create<UiEvent>()!!
```
[the remark this answers is not in the document]
Comment (saket) [reply app/src/main/java/org/simple/clinic/medicalhistory/MedicalHistoryQuestionView.kt]: Um… I'm not to keen about doing this. But considering that I'm making changes for this PR, I might as well do this.

**Comment by vinaysshenoy** [comment]
This was a suggestion, fyi.

**Comment by saket** [comment]
@vinaysshenoy I know and I also understood your reasoning behind not exposing the subject :)

**Pull request merged**